src/derivative.jl

```diff
@@ -0,0 +1,57 @@
+# This file is a part of LegendDSP.jl, licensed under the MIT License (MIT).
+
+"""
+    struct DerivativeFilter <: AbstractRadIIRFilter
+
+Working example:
+```julia
+using LegendDSP
+using Unitful
+
+signal = rand(100)
+t = range(0u"ms", 20u"ms", 100)
+
+wf = RDWaveform(t, signal)
+flt = DerivativeFilter()
+wf_new = flt(wf)
+```
+Constructors:
+
+* ```$(FUNCTIONNAME)(; fields...)```
+
+Fields:
+
+$(TYPEDFIELDS)
+"""
+Base.@kwdef struct DerivativeFilter{T <: RadiationDetectorDSP.RealQuantity} <: AbstractRadIIRFilter
+    "Filter gain"
+    gain::T = 1
+end
+export DerivativeFilter
+
+struct DerivativeFilterInstance{T <: RadiationDetectorDSP.RealQuantity, TG <: RadiationDetectorDSP.RealQuantity} <: AbstractRadSigFilterInstance{LinearFiltering}
+    gain::TG
+    n_input::Int
+end
+
+function RadiationDetectorDSP.fltinstance(flt::DerivativeFilter{TG}, si::SamplingInfo{T}) where {T, TG}
+    DerivativeFilterInstance{T, TG}(flt.gain, _smpllen(si))
+end
+
+
+RadiationDetectorDSP._filterlen(fi::DerivativeFilterInstance) = 1
```

Comment:
Shouldn't it be two? Or does it have to be one because `rdfilt` uses it to construct the correct output length?